nupic/research/frameworks/vernon/mixins/noise_robustness_test.py

```diff
@@ -0,0 +1,146 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import sys
+from functools import partial
+
+import torch
+import torch.nn.functional as F
+from torchvision import transforms
+
+from nupic.research.frameworks.pytorch.model_utils import evaluate_model
+
+
+class NoiseRobustnessTest:
+    """
+    Replaces the standard evaluate model function with a loop over the same function
+    that conducts a noise robustness test on each iteration. In the test, validation
+    accuracy is reported for samples with varying percentages of Gaussian
+    noise. The output of this mixin is a curve with noise percentage on the x axis
+    and validation accuracy on the y axis.
+    """
+
+    def setup_experiment(self, config):
+        """
+        :param config:
+            - noise_levels: Optional, a list of floats between 0 and 1 which specify
```

Comment:
1. You should also include `noise_std` and `noise_mean` in the docstring, since they're config params specific to this mixin.
2. It would be helpful to include an example config, just like how we did for the dendrite metrics mixin [here](https://github.com/numenta/nupic.research/blob/master/nupic/research/frameworks/vernon/mixins/dendrite_metrics.py#L65-L107).

Reply:
Done